### DISCUSSION TITLE
How to implement a hysteresis? I keep 70% but it turn on below it and turn of at 71%

### DISCUSSION THREAD

**Discussion by gasilva** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#222] [Ideas]
I keep 70% but it turn on below it and turn of at 71%

How can I have turn on at 30%
And turn off at 70%?

**Comment by danfoerster** [comment]
The battery will discharge very slowly on its own. If you unplug your Macbook at night, it could lose exactly one percent so it tops-off again when the charger is reconnected. You can set your max charge value, for example 70. Wait til the battery stops being charged and then set the value to 30. Then it will not be re-charged until it has dropped to 30%. But keep in mind, once it does it will only charge from 29 to 30% and then stop. You will have to change the value to 70 again.

**Reply by mikebore** [reply]
Presume this will only happen with the "Allow Discharge" setting off.

**Reply by danfoerster** [reply]
Yes I suppose. I have an Intel Mac with and the Classic SMC key mode enabled, so I don't have the discharge option to enable

**Reply by mikebore** [reply]
Thanks. I think this is a great way of using Al Dente. With my usage it is going to take a long time (days/weeks) to drop from 70 to 30, so not a chore to reset AlDente to 70, then back to 30. The end result is a battery that is only used in the range 30-70 with very low cycle rate increase. I think it has already been asked if this could be an automated sequence in Al Dente.
Occasional excursions to 100 and back to 0% seem to be recommended.

**Reply by danfoerster** [reply]
I think so as well. My Mac is plugged in all the time and barely ever moved from my desk. I just leave it plugged in now (also over night) and the battery by its self-discharge barely drops a percent in a week or so. I do a 100% to below 20 about once a month. Yes I think so too that batteries like to be used and its better to have them charge and discharge occasionally as opposed to only storing them without any 'exercise'
